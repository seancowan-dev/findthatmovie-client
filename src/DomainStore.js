import {observable,action,toJS} from 'mobx';
import uuid from "uuid";

class DomainStore{
    // App Primary 
     @observable updateType = "";
     @observable updateFolderId = "";
     @observable updateNoteId = "";
     @observable currentNote = null;
     @observable done = "";
     @observable updated = null;
     @observable errorMsg = "";
     @observable notes = {};
     @observable folders = {};

    // Update Component
     @observable folderNotes = [];
     @observable selectedFolder = "";
     @observable selectedFolderId = "";
     @observable selectedFolderIdOld = "";
     @observable oldNoteId = null;
     @observable currentNoteId = null;
     @observable migrateChecked = false;
     @observable note = {
         id: null,
         name: null,
         folderid: null,
         modified: null,
         content: null
     };

    // Local methods to use with fetch api methods
     handleErrors(response) { // prepares error message for HTTP request errors
      if (response.ok === true) {
          return response.json();
      } else {
          throw new Error("Code " + response.status + " Message: " + response.statusText)
      }
    }
    hex(string) {
      let source = unescape(encodeURIComponent(string));
      let hex = '';
        for (var i = 0; i < source.length; i++) {
            hex += source.charCodeAt(i).toString(16)
        };
        return hex;
    }

  // Fetch API Methods

  @action async getFolders() {
    return await fetch(`http://localhost:8000/api/folders`, {
      method: 'GET',
      headers: {
        'content-type': 'application/json'
      },
    })
    .then(response => this.handleErrors(response))
    .then(resJSON => { return resJSON })
    .catch(e => alert(e));
  }

  @action async getNotes() {
    return await fetch(`http://localhost:8000/api/notes`, {
      method: 'GET',
      headers: {
        'content-type': 'application/json'
      },
    })
    .then(response => this.handleErrors(response))
    .then(resJSON => { return resJSON })
    .catch(e => alert(e));
  }

  @action async deleteNote(id) {
    return await fetch(`http://localhost:8000/api/notes/${id}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      },
    })
    .then(response => { 
      if (response.ok === true) {
        this.promises();
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    })
    .catch(e => alert(e));

  }

  @action async addNote(event) {
    event.preventDefault();
    let workingArr = Array.from(event.target.parentNode.childNodes);
    let date = new Date(Date.now());
    let note = {
      id: uuid.v4(),
      name: "",
      modified: date.toISOString(),
      folderid: "",
      content: ""
    };
    workingArr.filter(child => {
      if(child.name === "add-note-content") {
        note.content = child.value;
      }
      if (child.name === "add-note-title") {
        note.name = child.value;
      }
      if (child.name === "select-note-folder") {
        note.folderid = child.selectedOptions[0].id;
      }
      return null
    });
    await fetch(`http://localhost:8000/api/notes/`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(note)
    })
    .then(response => {
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    }).catch(e => alert(e));    
  }

  @action async addFolder(event) {
    event.preventDefault();
    let workingArr = Array.from(event.target.parentNode.childNodes);
    let folder = {
      id: "",
      name: "",
    };
    workingArr.filter(child => {
      if(child.name === "add-folder-title") {
        folder.name = child.value;
        folder.id = uuid.v4();
      }
      return null
    });

    await fetch(`http://localhost:8000/api/folders/`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(folder)
    })
    .then(response => {
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    }).catch(e => alert(e));
  }

  @action async deleteFolder(id) {
    await fetch(`http://localhost:8000/api/folders/${id}`, {
      method: `DELETE`,
      headers: {
        'content-type': 'application/json'
      },

    })
    .then(response => { 
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    })
    .catch(e => alert(e));
  }

  @action async updateFolder(id, updates) {
    await fetch(`http://localhost:8000/api/folders/${id}`, {
      method: `PATCH`,
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(updates),
    })
    .then(response => {
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    })
    .catch(e => alert(e));
  }

  @action async updateNote(id, updates) {
    await fetch(`http://localhost:8000/api/notes/${id}`, {
      method: `PATCH`,
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(updates),
    })
    .then(response => {
        if (response.ok === true) {
          this.promises()
        } else {
          throw new Error("Code " + response.status + " Message: " + response.statusText)
        }
    })
    .catch(e => alert(e));
  }

  @action promises() {
    Promise.all([this.getFolders(), this.getNotes()]).then(res => {
      this.folders = res[0];
      this.notes = res[1];
      this.done = true;
    });
  }
}
export default new DomainStore();