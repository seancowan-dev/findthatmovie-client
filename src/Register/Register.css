.create-account-user-info {
    display: grid;
    font-size: 12px;
    grid-template-columns: 40% 40% auto;
    background-color: rgb(161, 161, 161);
    grid-gap: 3%;
    margin-bottom: 30px;
    height: 200px;
  }
  
  .create-account-form {
    background-color: rgb(110, 110, 110);
  }