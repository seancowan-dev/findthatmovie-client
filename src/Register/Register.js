import React from 'react';
import { observer, inject } from 'mobx-react';
import UserService from '../services/users-service';
import { navigate, A } from 'hookrouter';
import { Message } from 'semantic-ui-react';
import bcrypt from 'bcryptjs';
import moment from 'moment';
import $ from 'jquery';
import './Register.css';
import uuid from 'uuid';


const Register = inject('userStore', 'validators', 'helpers')(observer((props) => {
    
    return <div className="container">
        <form className="create-account-form">
            <fieldset className="create-account-user-info">
                <h1>Create New Account</h1>
                <Message key={uuid.v4()} floating className={props.helpers.checkMessageVisible(props.validators.registration.visible)} warning>
                    <Message.Header>{props.validators.registration.message}</Message.Header>
                    <Message.List items={Array.from(props.validators.arrayMsgs)} />
                </Message>
                    <input 
                        placeholder="Enter Username" 
                        className="username-input register-inputs inputs" 
                        type="text"
                        value={props.userStore.newRegistrant.username}
                        onClick={(e) => {
                            $('.MuiInput-input').removeAttr("readonly");
                        }}
                        onChange={(e) =>{
                            props.userStore.newRegistrant.username = e.target.value;
                        }}
                    />
                    <input 
                        placeholder="Enter E-mail" 
                        className="email-input register-inputs inputs" 
                        type="text"
                        value={props.userStore.newRegistrant.email}
                        onChange={(e) =>{
                            props.userStore.newRegistrant.email = e.target.value;
                        }}                        
                    />
                    <input
                        placeholder="Confirm E-mail" 
                        className="email-confirm-input register-inputs inputs" 
                        type="text"
                        value={props.userStore.newRegistrant.confirmEmail}
                        onChange={(e) =>{
                            props.userStore.newRegistrant.confirmEmail = e.target.value;
                        }}                            
                    />
                    <input 
                        placeholder="Enter Password" 
                        className="password-input register-inputs inputs" 
                        type="password"
                        value={props.userStore.newRegistrant.password}
                        onChange={(e) =>{
                            props.userStore.newRegistrant.password = e.target.value;
                        }}                            
                    />
                    <input 
                        placeholder="Confirm Password" 
                        className="password-confirm-input register-inputs inputs" 
                        type="password"
                        value={props.userStore.newRegistrant.confirmPassword}
                        onChange={(e) =>{
                            props.userStore.newRegistrant.confirmPassword = e.target.value;
                        }}                            
                    />
                    <button  
                    type="submit" 
                    className="register-submit buttons" 
                    onClick={(e) => {
                        e.preventDefault();

                        // Check Data
                        let data = {
                            userName: props.userStore.newRegistrant.username,
                            email: props.userStore.newRegistrant.email,
                            emailConfirm: props.userStore.newRegistrant.confirmEmail,
                            password: props.userStore.newRegistrant.password,
                            passwordConfirm: props.userStore.newRegistrant.confirmPassword
                        }

                        // Perform Validations
                        let validator_object = props.validators.registrationValidator(data);
                        
                        if (validator_object[0] === undefined) { 
                            // If a reference to index 0 of the array returns undefined then no errors were generated

                            // Just to be sure verify that valid was returned true
                            if (validator_object.valid === true) {
                                // If we reach this point we should create the user's account
                                bcrypt.hash(props.userStore.newRegistrant.password, 8, function(err, hash) {
                                    let confirmed = {
                                        "name": props.userStore.newRegistrant.username,
                                        "password": hash,
                                        "email": props.userStore.newRegistrant.email,
                                        "perm_level": "user",
                                        "created_at": moment().format()
                                    }
                                    UserService.addUser(JSON.stringify(confirmed));
                                    navigate("/login");
                                });
                            }

                        } else { // If we reach this point we should list all the errors for the user
                            props.validators.setRegistrationVisibility(true);
                            let arrayMsgs = [];
                            validator_object.map(item => {
                                arrayMsgs.push(item.message);
                                return null;
                            })
                            props.validators.setRegistrationMessage("We detected the following issues with your registration")
                            props.validators.setRegistrationMessageArray(arrayMsgs);
                        }
                    }}>Create</button>
                    <p>Already have an account? <A href="/login">Click here</A></p>
            </fieldset>
        </form>
    </div>
})); 

export default Register;