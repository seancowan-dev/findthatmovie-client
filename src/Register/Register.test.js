import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from "mobx-react";
import DomainStore from '../DomainStore';
import Register from './Register';

const store = {
    dataStore: DomainStore.dataStore,
    searchStore: DomainStore.searchStore,
    helpers: DomainStore.helpers,
    userStore: DomainStore.userStore,
    validators: DomainStore.validators
  }

describe('| Register Test Object |', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Provider {...store}><Register debug /></Provider>);
    expect(component).toMatchSnapshot();
  });
  it('should render correctly in "production" mode', () => {
    const component = shallow(<Provider {...store}><Register production /></Provider>);
    expect(component).toMatchSnapshot();
  });
});