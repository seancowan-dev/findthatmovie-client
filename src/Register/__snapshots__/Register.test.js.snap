// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`| Register Test Object | should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`| Register Test Object | should render correctly in "production" mode 1`] = `ShallowWrapper {}`;
